### ISSUE TITLE
[Question] Kafka Connect status topic is not updated

### ISSUE THREAD

**Issue by klalafaryan** [strimzi/strimzi-kafka-operator#3225]
Hey All,

We are running kafka connect cluster with strimzi.
When we deploy a new kafka connector, sometimes connector gets wrong `worker id` ( wordker ID with the IP which doesn't exist anymore, because of new deployment), and it is in `unassigned` state.

We have to manually delete `kafka connect statuses` topic  to be able to refresh the worker id. 

How can we tackle this case ? 

Any suggestions are appreciated.

Thanks.

**Comment by scholzj** [comment]
Kafka normally deals with this on its own. I have never seen this problem before.

**Comment by klalafaryan** [comment]
@scholzj  Thanks for your reply. We are trying to debug this, and seems like it is easy to reproduce.  We have noticed one more problem which maybe the root cause of the issue.

Sometimes we are getting following log message infinitely.

```
2020-06-23 08:28:16,619 INFO [Worker clientId=connect-1, groupId=group-id] Rebalance started (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator) [DistributedHerder-connect-1-1]
2020-06-23 08:28:16,619 INFO [Worker clientId=connect-1, groupId=group-id] (Re-)joining group (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-06-23 08:28:16,621 INFO [Worker clientId=connect-1, groupId=group-id] Was selected to perform assignments, but do not have latest config found in sync request. Returning an empty configuration to trigger re-sync. (org.apache.kafka.connect.runtime.distributed.IncrementalCooperativeAssignor) [DistributedHerder-connect-1-1]
2020-06-23 08:28:16,624 INFO [Worker clientId=connect-1, groupId=group-id] Successfully joined group with generation 22192 (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-06-23 08:28:16,624 INFO [Worker clientId=connect-1, groupId=group-id] Joined group at generation 22192 with protocol version 2 and got assignment: Assignment{error=1, leader='connect-1-10b7ab14-0cc9-4290-a72b-e73e5f8773ac', leaderUrl='http://10.112.17.144:8083/', offset=70, connectorIds=[], taskIds=[], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder) [DistributedHerder-connect-1-1]
```

Do you have some ideas ?

**Comment by scholzj** [comment]
Not really. It looks just like a regular chatter from the syncing of the Connect nodes. If the re-balance is happening too often, maybe there is some network issue which does not keep the connections stable?

**Comment by klalafaryan** [comment]
we have checked and seems like the network is fine. Still we couldn't figure out the problem.

But we did following:

1.  We have to kafka connect instances. We have executed `kubectl port-forward` into 8083:8083 and 8084:8083

When we execute `http://localhost:8084/connectors/source-mysql-example/status` (kafka-connect pod 1) we can see following response:
```
{"name":"source-mysql-example","connector":{"state":"RUNNING","worker_id":"10.112.44.132:8083"},"tasks":[],"type":"source"}
```

When we execute `http://localhost:8084/connectors/source-mysql-example/status` (kafka-connect pod 2) we can see following response:
```
{"name":"source-mysql-example","connector":{"state":"RUNNING","worker_id":"10.112.44.132:8083"},"tasks":[{"id":0,"state":"UNASSIGNED","worker_id":"10.112.17.144:8083"}],"type":"source"}
```
**_NOTE: as you can see IP address 10.112.17.144 is not available anymore this is the IP from the old kafka connect pod._**

So basically we are getting two different responses from two different pods in the same kafka connect cluster.

2.  If we try to restart the connector with `curl -X POST localhost:8084/connectors/source-mysql-example/restart`

We can see following messages have been produced into the `kafka-connect-statues` topic:

```
key status-connector-source-mysql-example/ value {"state":"UNASSIGNED","trace":null,"worker_id":"10.112.44.132:8083","generation":43087}
key status-connector-source-mysql-example/ value {"state":"RUNNING","trace":null,"worker_id":"10.112.44.132:8083","generation":43087}
```

Any ideas or directions how can we debug further ?
Thanks a lot.

**Comment by klalafaryan** [comment]
@scholzj  Do you have some ideas ? Do we need to follow up this with kafka community ?

And alos we are wondering how strimzi operator deploys kafka-connect and connectors, maybe something is wrong with the deployment order. (For example it first deploys connectors without waiting for kafka connect to be ready)   WDYT ?

**Comment by scholzj** [comment]
TBH, I'm still not sure I understand the problem really. So I'm not sure I can help.

**Comment by klalafaryan** [comment]
@scholzj  Sorry for confusion. 
So the main problem that we are having is the following infinite loop:

```
2020-07-09 08:51:25,731 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Rebalance started (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,731 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] (Re-)joining group (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,733 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Was selected to perform assignments, but do not have latest config found in sync request. Returning an empty configuration to trigger re-sync. (org.apache.kafka.connect.runtime.distributed.IncrementalCooperativeAssignor) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,735 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Successfully joined group with generation 305655831 (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,735 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Joined group at generation 305655831 with protocol version 2 and got assignment: Assignment{error=1, leader='connect-1-0008abc5-a152-42fe-a697-a4a4641f72bb', leaderUrl='http://10.60.42.221:8083/', offset=12, connectorIds=[], taskIds=[], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,735 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Rebalance started (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,735 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] (Re-)joining group (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,736 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Was selected to perform assignments, but do not have latest config found in sync request. Returning an empty configuration to trigger re-sync. (org.apache.kafka.connect.runtime.distributed.IncrementalCooperativeAssignor) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,739 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Successfully joined group with generation 305655832 (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,739 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Joined group at generation 305655832 with protocol version 2 and got assignment: Assignment{error=1, leader='connect-1-0008abc5-a152-42fe-a697-a4a4641f72bb', leaderUrl='http://10.60.42.221:8083/', offset=12, connectorIds=[], taskIds=[], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,739 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Rebalance started (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,739 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] (Re-)joining group (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,740 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Was selected to perform assignments, but do not have latest config found in sync request. Returning an empty configuration to trigger re-sync. (org.apache.kafka.connect.runtime.distributed.IncrementalCooperativeAssignor) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,742 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Successfully joined group with generation 305655833 (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,742 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Joined group at generation 305655833 with protocol version 2 and got assignment: Assignment{error=1, leader='connect-1-0008abc5-a152-42fe-a697-a4a4641f72bb', leaderUrl='http://10.60.42.221:8083/', offset=12, connectorIds=[], taskIds=[], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,742 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Rebalance started (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,742 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] (Re-)joining group (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,744 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Was selected to perform assignments, but do not have latest config found in sync request. Returning an empty configuration to trigger re-sync. (org.apache.kafka.connect.runtime.distributed.IncrementalCooperativeAssignor) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,746 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Successfully joined group with generation 305655834 (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,746 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Joined group at generation 305655834 with protocol version 2 and got assignment: Assignment{error=1, leader='connect-1-0008abc5-a152-42fe-a697-a4a4641f72bb', leaderUrl='http://10.60.42.221:8083/', offset=12, connectorIds=[], taskIds=[], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,746 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Rebalance started (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,746 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] (Re-)joining group (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,748 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Was selected to perform assignments, but do not have latest config found in sync request. Returning an empty configuration to trigger re-sync. (org.apache.kafka.connect.runtime.distributed.IncrementalCooperativeAssignor) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,750 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Successfully joined group with generation 305655835 (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,750 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Joined group at generation 305655835 with protocol version 2 and got assignment: Assignment{error=1, leader='connect-1-0008abc5-a152-42fe-a697-a4a4641f72bb', leaderUrl='http://10.60.42.221:8083/', offset=12, connectorIds=[], taskIds=[], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,750 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Rebalance started (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,750 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] (Re-)joining group (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,751 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Was selected to perform assignments, but do not have latest config found in sync request. Returning an empty configuration to trigger re-sync. (org.apache.kafka.connect.runtime.distributed.IncrementalCooperativeAssignor) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,754 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Successfully joined group with generation 305655836 (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,754 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Joined group at generation 305655836 with protocol version 2 and got assignment: Assignment{error=1, leader='connect-1-0008abc5-a152-42fe-a697-a4a4641f72bb', leaderUrl='http://10.60.42.221:8083/', offset=12, connectorIds=[], taskIds=[], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,754 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Rebalance started (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,754 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] (Re-)joining group (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,755 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Was selected to perform assignments, but do not have latest config found in sync request. Returning an empty configuration to trigger re-sync. (org.apache.kafka.connect.runtime.distributed.IncrementalCooperativeAssignor) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,757 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Successfully joined group with generation 305655837 (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,757 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Joined group at generation 305655837 with protocol version 2 and got assignment: Assignment{error=1, leader='connect-1-0008abc5-a152-42fe-a697-a4a4641f72bb', leaderUrl='http://10.60.42.221:8083/', offset=12, connectorIds=[], taskIds=[], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,757 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Rebalance started (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,757 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] (Re-)joining group (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,759 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Was selected to perform assignments, but do not have latest config found in sync request. Returning an empty configuration to trigger re-sync. (org.apache.kafka.connect.runtime.distributed.IncrementalCooperativeAssignor) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,761 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Successfully joined group with generation 305655838 (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,761 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Joined group at generation 305655838 with protocol version 2 and got assignment: Assignment{error=1, leader='connect-1-0008abc5-a152-42fe-a697-a4a4641f72bb', leaderUrl='http://10.60.42.221:8083/', offset=12, connectorIds=[], taskIds=[], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,761 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Rebalance started (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,761 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] (Re-)joining group (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,763 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Was selected to perform assignments, but do not have latest config found in sync request. Returning an empty configuration to trigger re-sync. (org.apache.kafka.connect.runtime.distributed.IncrementalCooperativeAssignor) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,766 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Successfully joined group with generation 305655839 (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,766 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Joined group at generation 305655839 with protocol version 2 and got assignment: Assignment{error=1, leader='connect-1-0008abc5-a152-42fe-a697-a4a4641f72bb', leaderUrl='http://10.60.42.221:8083/', offset=12, connectorIds=[], taskIds=[], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,766 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Rebalance started (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,766 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] (Re-)joining group (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,768 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Was selected to perform assignments, but do not have latest config found in sync request. Returning an empty configuration to trigger re-sync. (org.apache.kafka.connect.runtime.distributed.IncrementalCooperativeAssignor) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,771 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Successfully joined group with generation 305655840 (org.apache.kafka.clients.consumer.internals.AbstractCoordinator) [DistributedHerder-connect-1-1]
2020-07-09 08:51:25,771 INFO [Worker clientId=connect-1, groupId=datalake-kafka-connect] Joined group at generation 305655840 with protocol version 2 and got assignment: Assignment{error=1, leader='connect-1-0008abc5-a152-42fe-a697-a4a4641f72bb', leaderUrl='http://10.60.42.221:8083/', offset=12, connectorIds=[], taskIds=[], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder) [DistributedHerder-connect-1-1]
```

**Comment by klalafaryan** [comment]
And in the broker side we can see following:
```
2020-07-09 16:39:46,260 INFO [GroupCoordinator 0]: Preparing to rebalance group kafka-connect in state PreparingRebalance with old generation 311127279 (__consumer_offsets-7) (reason: Updating metadata for member connect-1-bdf1c132-3311-493a-894a-5fffcac41ec7) (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-0]
2020-07-09 16:39:46,261 INFO [GroupCoordinator 0]: Stabilized group kafka-connect generation 311127280 (__consumer_offsets-7) (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-5]
2020-07-09 16:39:46,262 INFO [GroupCoordinator 0]: Assignment received from leader for group kafka-connect for generation 311127280 (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-1]
2020-07-09 16:39:46,265 INFO [GroupCoordinator 0]: Preparing to rebalance group kafka-connect in state PreparingRebalance with old generation 311127280 (__consumer_offsets-7) (reason: Updating metadata for member connect-1-bdf1c132-3311-493a-894a-5fffcac41ec7) (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-1]
2020-07-09 16:39:46,266 INFO [GroupCoordinator 0]: Stabilized group kafka-connect generation 311127281 (__consumer_offsets-7) (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-6]
2020-07-09 16:39:46,267 INFO [GroupCoordinator 0]: Assignment received from leader for group kafka-connect for generation 311127281 (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-1]
2020-07-09 16:39:46,270 INFO [GroupCoordinator 0]: Preparing to rebalance group kafka-connect in state PreparingRebalance with old generation 311127281 (__consumer_offsets-7) (reason: Updating metadata for member connect-1-bdf1c132-3311-493a-894a-5fffcac41ec7) (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-7]
2020-07-09 16:39:46,271 INFO [GroupCoordinator 0]: Stabilized group kafka-connect generation 311127282 (__consumer_offsets-7) (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-6]
2020-07-09 16:39:46,272 INFO [GroupCoordinator 0]: Assignment received from leader for group kafka-connect for generation 311127282 (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-1]
2020-07-09 16:39:46,275 INFO [GroupCoordinator 0]: Preparing to rebalance group kafka-connect in state PreparingRebalance with old generation 311127282 (__consumer_offsets-7) (reason: Updating metadata for member connect-1-bdf1c132-3311-493a-894a-5fffcac41ec7) (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-3]
2020-07-09 16:39:46,276 INFO [GroupCoordinator 0]: Stabilized group kafka-connect generation 311127283 (__consumer_offsets-7) (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-7]
2020-07-09 16:39:46,277 INFO [GroupCoordinator 0]: Assignment received from leader for group kafka-connect for generation 311127283 (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-5]
2020-07-09 16:39:46,280 INFO [GroupCoordinator 0]: Preparing to rebalance group kafka-connect in state PreparingRebalance with old generation 311127283 (__consumer_offsets-7) (reason: Updating metadata for member connect-1-bdf1c132-3311-493a-894a-5fffcac41ec7) (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-5]
2020-07-09 16:39:46,281 INFO [GroupCoordinator 0]: Stabilized group kafka-connect generation 311127284 (__consumer_offsets-7) (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-7]
2020-07-09 16:39:46,282 INFO [GroupCoordinator 0]: Assignment received from leader for group kafka-connect for generation 311127284 (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-3]
2020-07-09 16:39:46,285 INFO [GroupCoordinator 0]: Preparing to rebalance group kafka-connect in state PreparingRebalance with old generation 311127284 (__consumer_offsets-7) (reason: Updating metadata for member connect-1-bdf1c132-3311-493a-894a-5fffcac41ec7) (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-1]
2020-07-09 16:39:46,286 INFO [GroupCoordinator 0]: Stabilized group kafka-connect generation 311127285 (__consumer_offsets-7) (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-4]
2020-07-09 16:39:46,287 INFO [GroupCoordinator 0]: Assignment received from leader for group kafka-connect for generation 311127285 (kafka.coordinator.group.GroupCoordinator) [data-plane-kafka-request-handler-7]
```

**Comment by scholzj** [comment]
That looks like something is connecting and disconnecting with the same consumer group `kafka-connect` and triggering the rebalance. But I have no idea what could cause it I'm afraid.

**Comment by klalafaryan** [comment]
@scholzj  Thanks for your reply. 
We have the same problem with the kafka connect cluster which we have deployed without strimzi. 
Seems like it is not related to strimzi, so I have created a bug ticket for kafka connect:
https://issues.apache.org/jira/browse/KAFKA-10253?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=17154790#comment-17154790

**Comment by klalafaryan** [comment]
Seems like we have found a problem, and it is related to the `config.storage.topic` topic which has been created with 3 partitions but it should be created always with **_single_** partition.
 
config.storage.topic
The name of the topic where connector and task configuration data are stored. This must be the same for all Workers with the same group.id. Kafka Connect will upon startup attempt to automatically create this topic with a single-partition and compacted cleanup policy to avoid losing data, but it will simply use the topic if it already exists. If you choose to create this topic manually, always create it as a compacted topic **_with a single partition_** and a high replication factor (3x or more).

So closing this ticket.
